### PR TITLE
SPM-1261: new endpoint for deleting a baseline

### PR DESCRIPTION
**Pull request by mkholjuraev**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: database_admin/migrations/073_modify_system_platform_permissions.up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+GRANT SELECT, UPDATE, DELETE, INSERT ON system_platform TO manager;
```
Comment (Josca) [inline database_admin/migrations/073_modify_system_platform_permissions.up.sql:1]: SQL script ...down. should be filled up accordingly.

File: manager/routes/routes.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ func InitAPI(api *gin.RouterGroup) {
 	baselines := api.Group("/baselines")
 	baselines.GET("/", controllers.BaselinesListHandler)
 	baselines.GET("/:baseline_id/systems", controllers.BaselineSystemsListHandler)
+	baselines.DELETE("/:baseline_id", controllers.BaselineDeleteHandler)
```
Comment (Josca) [inline manager/routes/routes.go:23]: I think write rbac permission should be used. So we need to update rbac.go.

File: base/database/testing.go [hunk 3]
```diff
@@ -312,3 +313,38 @@ func UpdateSystemAdvisoriesWhenPatched(t *testing.T, systemID, accountID int, ad
...
+
+	baselineConfig, err := json.Marshal(BaselineConfig{ToTime: "2021-01-01 12:00:00-04"})
+	assert.Nil(t, err)
+
+	temporaryBaseline := &models.Baseline{
+		RhAccountID: 1, Name: "temporary_baseline", Config: baselineConfig,
+	}
+	err = Db.Create(temporaryBaseline).Error
+	assert.Nil(t, err)
+
```
Comment (Josca) [inline base/database/testing.go:330]: We should ensure the db row was really added.

**Review by Josca — CHANGES_REQUESTED** [review]
Thanks, just some todos added.

File: manager/routes/routes.go (hunk 2 above)
Comment (mkholjuraev) [reply manager/routes/routes.go:23]: I believe we have already permission to delete: https://github.com/RedHatInsights/patchman-engine/blob/master/manager/middlewares/rbac.go#L77

**Comment by Josca** [comment]
Closing, incloded to #886

**Pull request closed without merging**